### PR TITLE
FastAPI Framework Setup

### PR DESCRIPTION
**Pull request by XXXJumpingFrogXXX**
This PR includes the following changes:
(1) Created the latest main.py file and completed some basic FastAPI settings in it.
(2) Renamed the original main.py file to original_main.py.
(3) Kept the existing piggy directory and created a chat directory to establish separate routers and APIs for each project.

### CODE REVIEW

File: chat/router.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+from fastapi import APIRouter
```
Comment (chimosky) [inline chat/router.py:1]: This is imported but unused.

File: chat/router.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    f'''
+                    Your task is to answer children's questions using simple language.
+                    Explain any difficult words in a way a 3-year-old can understand.
+                    Keep responses under 60 words.
+                    \n\nQuestion: {value.query}
+                    ''',  # instruction
+                    "",  # input
+                    "",  # output - leave this blank for generation!
+                )
+            ], return_tensors="pt").to("cuda")
```
Comment (chimosky) [inline chat/router.py:58]: These should be assigned once and used whenever as it seems their params won't change after the first assignment, there's no reason to reassign them whenever the function is called.

File: chat/router.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+### Input:
+{}
+
+### Response:
+{}"""
+
+class Question(BaseModel):
+    query: str
+
+@router.post("/generate_answer")
```
Comment (chimosky) [inline chat/router.py:25]: `router` doesn't exist yet, how come you're using it?

**Comment by chimosky** [comment]
Reviewed ce06570, your commit message should be improved to explain what exactly we're doing with FastAPI and why.

The header is just a summary, your change needs more than a summary.

File: chat/router.py (hunk 1 above)
Comment (XXXJumpingFrogXXX) [reply chat/router.py:1]: I modified the code in PR # 9. This is used in the new code and essential for fastapi part.

File: chat/router.py (hunk 3 above)
Comment (XXXJumpingFrogXXX) [reply chat/router.py:25]: Here's the above question. I have modified that.

**Comment by XXXJumpingFrogXXX** [comment]
> Reviewed [ce06570](https://github.com/sugarlabs/sugar-ai/commit/ce0657020b32065176e78fbe3f02cb9c34f152b7), your commit message should be improved to explain what exactly we're doing with FastAPI and why.
> 
> The header is just a summary, your change needs more than a summary.
> 
> Also, please avoid opening PRs with your master branch, checkout our [contributing](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#contributing) docs.

I agree. I make a detailed introduction in the new PR #9

**Comment by XXXJumpingFrogXXX** [comment]
I made the latest modifications in PR #9 and provided a detailed introduction, so I will close this PR.

**Pull request closed without merging**